### PR TITLE
fix: year zero was not accepted for SpannerDate

### LINKED ISSUE #782: Constructing a SpannerDate with year=0 fails
Creating a `SpannerDate` with y/m/d values, where year is 0 (zero) will return an invalid date:

```typescript
// The following will return a 'NaN-NaN-NaN' date.
const date = new codec.SpannerDate(0, 2, 22);
```

### PR DESCRIPTION
**Pull request by olavloite**
Creating a SpannerDate with year/month/date values with year=0 would return an invalid date.

Fixes #782

### CODE REVIEW

File: test/codec.ts [hunk 1]
```diff
@@ -75,6 +75,14 @@ describe('codec', () => {
         assert.strictEqual(json, '1986-03-22');
       });
 
+      it('should accept year zero in y/m/d number values', () => {
+        const d = new codec.SpannerDate(null!);
```
Comment (AVaksman) [inline test/codec.ts]: is value planned to be used?

File: src/codec.ts [hunk 2]
```diff
@@ -58,8 +58,9 @@ type DateFields = [number, number, number];
  * @extends Date
  *
  * @param {string|number} [date] String representing the date or number
- *     representing the year.
- * @param {number} [month] Number representing the month.
+ *     representing the year. If year is a number between 0 and 99, then year is
+ *     is assumed to be 1900 + year.
```
Comment (AVaksman) [inline src/codec.ts]: `If the year` ... `then the year is`
an extra `is` on line 62

**Comment by AVaksman** [comment]
LGTM

File: test/codec.ts [hunk 3]
```diff
@@ -75,6 +75,14 @@ describe('codec', () => {
...
+        const date = new codec.SpannerDate(0, 2, 22);
```
Comment (hengfengli) [inline test/codec.ts]: why is `(0, 2, 22)` instead of `(0, 3, 22)`? did this test pass? `json` should not be equal to `1900-03-22`.

Comment (AVaksman) [reply test/codec.ts]: The month are 0 based index ([link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getMonth/))
0 = January

Comment (hengfengli) [reply test/codec.ts]: Thanks. Got it. The built-in `Date` uses zero-based index. This is awful, 😭.

File: test/codec.ts (hunk 1 above)
Comment (olavloite) [reply test/codec.ts]: I added the missing assert.

File: src/codec.ts (hunk 2 above)
Comment (olavloite) [reply src/codec.ts]: Fixed.

**Pull request merged**